.github/workflows/build-test-lint-publish.yml

```diff
@@ -104,12 +104,41 @@ jobs:
           npm ci --legacy-peer-deps
           npm run build
 
-      - name: Publish the package
+      - name: Publish the package to npm
         id: publish
         uses: JS-DevTools/npm-publish@v1
         with:
           token: ${{ env.NPM_TOKEN }}
-
-      - if: steps.publish.outputs.type != 'none'
-        run: |
-          echo "Version changed: ${{ steps.publish.outputs.old-version }} => ${{ steps.publish.outputs.version }}"
+          # only publish if the version in package.json has changed
+          check-version: true
+
+      - name: Add git tag ${{ steps.publish.outputs.version }}
+        if: steps.publish.outputs.type != 'none'
+        uses: actions/github-script@v5
+        env:
+          NEW_VERSION: ${{ steps.publish.outputs.version }}
+          OLD_VERSION: ${{ steps.publish.outputs.old-version }}
+        with:
+          script: |
```

Comment:
Does yaml allow for importing scripts? 
Would it be better to have all this in a script and then import that over here?

Reply:
Yes, I think it's possible this could be imported as a script but I think I'd prefer not to abstract this away in a script.  I like that we can see exactly what's being done right here in the context of the workflow.  What would be more idiomatic if we did want to abstract this out or re-use it would be to make it a GitHub Action that we would specify in the `uses` property in order to import it.  Perhaps if this were really long and complicated that would be worth it but its short and simple enough that I think it's fine staying here.

Reply:
@siddharthpant92 - within reason, i generally follow the rule of three - if you do something three times, start thinking about automating it (in this context, making a GitHub Action). a one-off is fine, and i agree - abstracting into a script may obfuscate some of the functionality